merenlab/anvio
# News modal

you __will__ be made aware of anvi'o news, and you'll __like__ it!

## the change
The news tab is gone! 'What news tab?', you say? Exactly! Anvi'o news now lives in a modal popup. An unavoidable, attention-grabbing modal. Don't worry, we still use a cookie to track whether you've already read the news. We seek not to annoy, simply to _inform_!

## how it works
Anvi'o stores a cookie in the browser that tracks the last-read news item. If there's a newer item, or if the cookie is trashed, the user will see all _new_ news items, as denoted by a lovely blue orb. The user can select 'mark all as read', and the cycle will begin anew. Should the user wish to review already-seen news, they can access _all_ news items from the interactive interface's hamburger dropdown menu, affectionately known as __Anvi'o UI Jail__ . 

## to test 
from this branch, start a new interactive session in the browser. Nothing pops up, right? Try deleting the cookie (I use the `EditThisCookie` chrome extension). Refresh your window, boom! You should see all the news. Mark all as read and then head into the codebase to `anvio/data/interactive/js/news.js` and uncomment lines `15 - 20`. Save and refresh your browser. This will register a new mock news item, which should appear all by itself. 'Mark all as read' again, and you're all set.  

## note
Once this is approved for merging, let me know and I'll remove the test code from the branch :)
